src/Microsoft.Diagnostics.Runtime/src/DataTargets/DefaultDacLocator.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Diagnostics.Runtime
+{
+    internal class DefaultDacLocator : IDacLocator
+    {
+        private readonly DataTarget _dataTarget;
+
+        public DefaultDacLocator(DataTarget dataTarget)
+        {
+            _dataTarget = dataTarget ?? throw new ArgumentNullException(nameof(dataTarget));
+        }
+
+        public string FindDac(ClrInfo clrInfo)
+        {
+            if (clrInfo == null) throw new ArgumentNullException(nameof(clrInfo));
+
+            string moduleDirectory = Path.GetDirectoryName(clrInfo.ModuleInfo.FileName);
+            if (moduleDirectory == null)
```

Comment:
This shouldn't happen, we use the FIleName to create clrInfo right?  More importantly we shouldn't throw here.  Theoretically we may still find the right dac in _dataTarget.SymbolLocator.FindBinary.